component/pyroscope/ebpf/ebpf_linux.go

```diff
@@ -0,0 +1,229 @@
+		appendable:   flowAppendable,
+		args:         args,
+		targetFinder: tf,
+		session:      session,
+		argsUpdate:   make(chan Arguments),
+	}
+	res.updateTargetFinder()
+	return res, nil
+}
+
+type Arguments struct {
+	ForwardTo            []pyroscope.Appendable `river:"forward_to,attr"`
+	Targets              []discovery.Target     `river:"targets,attr,optional"`
+	DefaultTarget        discovery.Target       `river:"default_target,attr,optional"`
+	TargetsOnly          bool                   `river:"targets_only,attr,optional"`
+	CollectInterval      time.Duration          `river:"collect_interval,attr,optional"`
+	SampleRate           int                    `river:"sample_rate,attr,optional"`
+	PidCacheSize         int                    `river:"pid_cache_size,attr,optional"`
+	BuildIDCacheSize     int                    `river:"build_id_cache_size,attr,optional"`
+	SameFileCacheSize    int                    `river:"same_file_cache_size,attr,optional"`
+	CacheRounds          int                    `river:"cache_rounds,attr,optional"`
+	ContainerIDCacheSize int                    `river:"container_id_cache_size,attr,optional"`
+	CollectUserProfile   bool                   `river:"collect_user_profile,attr,optional"`
+	CollectKernelProfile bool                   `river:"collect_kernel_profile,attr,optional"`
+}
+
+func (rc *Arguments) UnmarshalRiver(f func(interface{}) error) error {
+	*rc = defaultArguments()
+	type config Arguments
+	return f((*config)(rc))
+}
+
+func defaultArguments() Arguments {
+	return Arguments{
+		CollectInterval:      10 * time.Second,
+		SampleRate:           100,
+		PidCacheSize:         32,
+		ContainerIDCacheSize: 1024,
+		BuildIDCacheSize:     64,
+		SameFileCacheSize:    8,
+		CacheRounds:          3,
+		TargetsOnly:          true,
+		CollectUserProfile:   true,
+		CollectKernelProfile: true,
+	}
+}
+
+type Component struct {
+	options      component.Options
+	args         Arguments
+	argsUpdate   chan Arguments
+	appendable   *pyroscope.Fanout
+	targetFinder *sd.TargetFinder
+	session      *ebpfspy.Session
+}
+
+func (c *Component) Run(ctx context.Context) error {
+	err := c.session.Start()
+	if err != nil {
+		return fmt.Errorf("ebpf profiling session start: %w", err)
+	}
+	defer c.session.Stop()
+
+	var g run.Group
+	g.Add(func() error {
+		collectInterval := c.args.CollectInterval
+		t := time.NewTicker(collectInterval)
+		defer t.Stop()
+		for {
+			select {
+			case <-ctx.Done():
+				return nil
+			case newArgs := <-c.argsUpdate:
+				c.args = newArgs
+				c.updateTargetFinder()
+				c.session.UpdateCacheOptions(cacheOptionsFromArgs(c.args))
+				err := c.session.UpdateSampleRate(c.args.SampleRate)
+				if err != nil {
+					return nil
+				}
+				c.appendable.UpdateChildren(newArgs.ForwardTo)
+				if c.args.CollectInterval != collectInterval {
+					t.Reset(c.args.CollectInterval)
+					collectInterval = c.args.CollectInterval
+				}
+			case <-t.C:
+				var err error
+				pprof.Do(
```

Comment:
Can you comment what this does and why it's required ?

Reply:
we don't need it. removed it.

it was just handy to separate ebpf.reset profile from the rest (mostly service discovery) during development